### ISSUE TITLE
Question about reproducing the results

### ISSUE THREAD

**Issue by e0jun** [google-research/deep_representation_one_class#2]
Thanks for your great work. 

I ran your 'contrastive_da.sh' file with

DATA=cifar10ood
SEED=3
CATEGORY=0

But I got 

{
    "embed.auc": 63.617005555555565,
    "embed.gde": 61.748266666666666,
    "embed.kde": 64.29441111111112,
    "embed.kocsvm": 63.649888888888896,
    "embed.locsvm": 66.37852222222223,
    "pool.auc": 77.99122777777778,
    "pool.gde": 80.53367777777778,
    "pool.kde": 74.89410000000001,
    "pool.kocsvm": 75.82452222222223,
    "pool.locsvm": 67.06372222222222
}.

And with other seed 72

I got 

{
    "embed.auc": 50.0,
    "embed.gde": 49.977777777777774,
    "embed.kde": 50.0,
    "embed.kocsvm": 49.977777777777774,
    "embed.locsvm": 50.022222222222226,
    "pool.auc": 60.998122222222214,
    "pool.gde": 60.27448888888889,
    "pool.kde": 56.09522222222223,
    "pool.kocsvm": 55.180677777777774,
    "pool.locsvm": 51.08432222222222
}

Could you let me know how to reproduce your results?! 
Thank you:)

**Comment by kihyuks** [comment]
Thanks @YoungjunLee-DM for trying our code out. 

I am not entirely sure what has happened. I am re-running on my end to see if I can reproduce your results. In the meantime, would you mind sharing `hparams.json` file, which you can find from [here](https://github.com/google-research/deep_representation_one_class/blob/a36feaa2e2f1f8d4aabe24d219e2220b187bd315/util/train.py#L435)?

**Comment by e0jun** [comment]
> Thanks @YoungjunLee-DM for trying our code out.
> 
> I am not entirely sure what has happened. I am re-running on my end to see if I can reproduce your results. In the meantime, would you mind sharing `hparams.json` file, which you can find from [here](https://github.com/google-research/deep_representation_one_class/blob/a36feaa2e2f1f8d4aabe24d219e2220b187bd315/util/train.py#L435)?

I am really thank you for your reply!
Below is the content of hparams.json file.
(I modified only seed when I ran 'run_contrastive_da.sh'.)

{
    "aug_list": "cnr0.5+hflip+jitter_b0.4_c0.4_s0.4_h0.4+gray0.2+blur0.5,+",
    "aug_list_for_test": "x",
    "batch_size": 32,
    "category": "0",
    "ckpt_epoch": 32,
    "ckpt_prefix": "",
    "dataset": "cifar10ood",
    "distaug_type": "rot",
    "file_path": "cifar10ood__s72_c0",
    "force_init": false,
    "head_dims": "512,512,512,512,512,512,512,512,128",
    "input_shape": "32,32,3",
    "is_validation": false,
    "latent_dim": 0,
    "learning_rate": 0.01,
    "method": "Contrastive",
    "model_dir": ".",
    "momentum": 0.9,
    "nesterov": false,
    "net_type": "ResNet18",
    "net_width": 1.0,
    "num_batch": 0,
    "num_epoch": 2048,
    "optim_type": "sgd",
    "regularize_bn": false,
    "root": "",
    "sched_freq": "epoch",
    "sched_gamma": 0.995,
    "sched_level": 7,
    "sched_min_rate": 0.0,
    "sched_step_size": 1,
    "sched_type": "cos",
    "seed": 72,
    "temperature": 0.2,
    "weight_decay": 0.0003
}

I just saw this file, is it right that "regularize_bn" is False?
Thank you:)

**Comment by kihyuks** [comment]
Hi @YoungjunLee-DM 

I've ran experiments from my end with SEED=1,3,72 and these are results:

SEED=1
{
    "embed.auc": 75.89605555555556,
    "embed.gde": 74.29400000000001,
    "embed.kde": 78.24138888888889,
    "embed.kocsvm": 77.32352222222222,
    "embed.locsvm": 74.83734444444444,
    "pool.auc": 90.1147888888889,
    "pool.gde": 91.0614,
    "pool.kde": 91.02767222222222,
    "pool.kocsvm": 90.89739999999999,
    "pool.locsvm": 87.4934
}

SEED=3
{
    "embed.auc": 77.26396666666666,
    "embed.gde": 75.70972222222223,
    "embed.kde": 79.97637777777778,
    "embed.kocsvm": 79.15825555555556,
    "embed.locsvm": 75.79307777777777,
    "pool.auc": 90.22420555555556,
    "pool.gde": 91.17758888888889,
    "pool.kde": 91.57093333333333,
    "pool.kocsvm": 91.4143888888889,
    "pool.locsvm": 88.70495555555556
}

SEED=72
{
    "embed.auc": 76.72969444444445,
    "embed.gde": 75.22174444444445,
    "embed.kde": 79.4802888888889,
    "embed.kocsvm": 78.70282222222222,
    "embed.locsvm": 78.52576666666667,
    "pool.auc": 89.88556666666668,
    "pool.gde": 90.85854444444443,
    "pool.kde": 91.34525555555555,
    "pool.kocsvm": 91.17268888888887,
    "pool.locsvm": 86.85264444444445
}

and this is hparams.json

{
    "aug_list": "cnr0.5+hflip+jitter_b0.4_c0.4_s0.4_h0.4+gray0.2+blur0.5,+",
    "aug_list_for_test": "x",
    "batch_size": 32,
    "category": "0",
    "ckpt_epoch": 32,
    "ckpt_prefix": "",
    "dataset": "cifar10ood",
    "distaug_type": "rot",
    "file_path": "cifar10ood__s72_c0",
    "force_init": false,
    "head_dims": "512,512,512,512,512,512,512,512,128",
    "input_shape": "32,32,3",
    "is_validation": false,
    "latent_dim": 0,
    "learning_rate": 0.01,
    "method": "Contrastive",
    "model_dir": ".",
    "momentum": 0.9,
    "nesterov": false,
    "net_type": "ResNet18",
    "net_width": 1.0,
    "num_batch": 0,
    "num_epoch": 2048,
    "optim_type": "sgd",
    "regularize_bn": false,
    "root": "",
    "sched_freq": "epoch",
    "sched_gamma": 0.995,
    "sched_level": 7,
    "sched_min_rate": 0.0,
    "sched_step_size": 1,
    "sched_type": "cos",
    "seed": 72,
    "temperature": 0.2,
    "weight_decay": 0.0003
}

I don't seem to identify any difference in hparams.json of yours and mine, so could be that the issue comes from compute environment. Did you happen to train a model using multiple GPUs? All my experiments are done using a single GPU and there might be some trouble if using multiple GPUs (haven't rigorously checked multiple GPU support).

**Comment by e0jun** [comment]
Thanks for your reply! @kihyuks 
As you guessed, I had ran it with multiple GPUs (two 2080ti). And when I've ran it with a single GPU, I can get the same results with you. That issue might come from using multiple GPUs. If you have time later, I'd really appreciate it if you could check multiple GPU support.
(I'm very new to this field, it was really good for me to learn a lot from your paper)

Best regards
